index.bs

```diff
@@ -271,7 +311,18 @@ interface FileSystemFileHandle : FileSystemHandle {
 };
 </xmp>
 
-A {{FileSystemFileHandle}}'s associated [=FileSystemHandle/entry=] must be a [=file entry=].
+A {{FileSystemFileHandle}}'s associated [=FileSystemHandle/path=]
+is expected to [=locate=] a [=file entry=].
```

Comment:
So, a `FileSystemDirectoryHandle` can also locate to a file entry. It just means that it shouldn't be used... for example:
```
let dirHandle = await root.getDirectoryHandle('test', { create: true });

await root.remove({ recursive: true });

let fileHandle = await root.getFileHandle('test', {create:true});
```
After these statements, `dirHandle` and `fileHandle` both correspond to the file "test", though operations (at least those which check the file system) on `dirHandle` will fail because the entry is the wrong type

I thought about overriding the "locate" algorithm for both `FileSystemDirectoryHandle` and `FileSystemFileHandle` to only return an entry if it's the correct type. Something like 
- let |result| be the result of locating the path
- return null if |result| is null
- return |result| if it's the correct type
- otherwise return "not a {file|directory}" 

And then `TypeMismatchError` can be thrown by the caller if it gets the "not a blah" string... but it seemed more clear (and provides a bit more flexibility) to just always check at the call site like:
```
  1. If |entry| is not a [=file entry=] [=/reject=] |result| with a
     "{{TypeMismatchError}}" {{DOMException}} and abort.
```

Thoughts?

Reply:
If a path is a list of strings, how do we know it' a directory? Would the last string be the empty string? (That's how the URL Standard does it, although it doesn't really deal in directories per se.)

Because if we did that locating would either return null or the correct entry, right?

Reply:
Added a "kind" to a "locator", which can now be a "file locator" or a "directory locator"